subsys/net/lib/app/CMakeLists.txt

```diff
@@ -6,5 +6,5 @@ zephyr_library_compile_definitions_ifdef(
 zephyr_library_sources_ifdef(CONFIG_NET_APP_SERVER net_app.c server.c)
 zephyr_library_sources_ifdef(CONFIG_NET_APP_CLIENT net_app.c client.c)
 
-zephyr_link_interface_ifdef(CONFIG_MBEDTLS mbedTLS)
+#zephyr_link_interface_ifdef(CONFIG_MBEDTLS mbedTLS)
```

Comment:
Why comment here, just remove the line if not used.

Reply:
As you could see, there was some work and discussion to get the changes right on the CMake side. Now that suitable solution is found, intermediate changes with comments, etc. are gone.